src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -99,11 +99,20 @@ MulticopterAttitudeControl::parameters_updated()
 	// Store some of the parameters in a more convenient way & precompute often-used values
 	_attitude_control.setProportionalGain(Vector3f(_param_mc_roll_p.get(), _param_mc_pitch_p.get(), _param_mc_yaw_p.get()));
 
-	// rate gains
-	_rate_p = Vector3f(_param_mc_rollrate_p.get(), _param_mc_pitchrate_p.get(), _param_mc_yawrate_p.get());
-	_rate_i = Vector3f(_param_mc_rollrate_i.get(), _param_mc_pitchrate_i.get(), _param_mc_yawrate_i.get());
+	// Rate gains
+	// The controller gain K is used to convert the parallel (P + I/s + sD) form
+	// to the ideal (K * [1 + 1/sTi + sTd]) form
+	const float p_k = _param_mc_rollrate_k.get();
+	const float q_k = _param_mc_pitchrate_k.get();
+	const float r_k = _param_mc_yawrate_k.get();
```

Comment:
It would improve readability to use these gains in `MulticopterAttitudeControl::control_attitude_rates() `, where the PID is implemented.

Reply:
Maybe TPA can scale the _K gains instead of the _P, _I and _D gains?

Reply:
I don't know for TPA, I'm not sure what's its real purpose. 
If it's just pure gain scheduling because the dynamic of the vehicle changes at high thrust, we should leave it as is and if it's due to thrust non-linearity, we should scale K.

Reply:
I am not sure either:
- The [documentation](https://docs.px4.io/master/en/config_mc/pid_tuning_guide_multicopter.html#thrust_curve) suggests that it corrects thrust non-linearity (suggesting TPA should be equal for P I and D components).
- However a [search on the codebase](https://github.com/PX4/Firmware/search?q=tpa&unscoped_q=tpa) shows that the platforms that use TPA (s250aq, zmr250, and qav250) do not scale the P I and D the same way. Maybe scaling K would have been equivalent?

For reference here is the PR that implements per-component TPA: https://github.com/PX4/Firmware/pull/5861